fracmanager/lifecycle_manager.go

```
@@ -0,0 +1,208 @@
+// Freezes writes, waits for pending operations, then seals the fraction.
+func (lc *lifecycleManager) Seal(active *activeProxy) error {
+	now := time.Now()
+	sealed, err := lc.provider.Seal(active.instance)
+	if err != nil {
+		return err
+	}
+	sealsTotal.Inc()
+	sealingTime := time.Since(now)
+	sealsDoneSeconds.Observe(sealingTime.Seconds())
+
+	logger.Info(
+		"fraction sealed",
+		zap.String("fraction", filepath.Base(sealed.BaseFileName)),
+		zap.Float64("time_spent_s", util.DurationToUnit(sealingTime, "s")),
+	)
+
+	lc.infoCache.Add(sealed.Info())
+	lc.registry.PromoteToLocal(active, sealed)
+	active.proxy.Redirect(sealed)
+	active.instance.Release()
+	return nil
+}
+
+// RotateIfNeeded checks if active fraction needs rotation based on size limit
+// Creates new active fraction and starts sealing the previous one.
+func (lc *lifecycleManager) Rotate(sizeLimit uint64, wg *sync.WaitGroup) {
+	if lc.registry.Active().instance.Info().DocsOnDisk > sizeLimit {
+		active := lc.rotate()
+
+		wg.Add(1)
+		lc.sealingWg.Add(1)
+		go func() {
+			defer wg.Done()
+			defer lc.sealingWg.Done()
+			if err := lc.Seal(active); err != nil {
+				logger.Fatal("sealing error", zap.Error(err))
+			}
+		}()
+	}
+}
+
+func (lc *lifecycleManager) rotate() *activeProxy {
+	active, err := lc.registry.Rotate(newActiveProxy(lc.provider.CreateActive()))
+	if err != nil {
+		logger.Fatal("active fraction rotation error", zap.Error(err))
+	}
+	return active
+}
+
+// OffloadLocal starts offloading of local fractions to remote storage
+// Selects fractions based on disk space usage and retention policy.
+func (lc *lifecycleManager) OffloadLocal(ctx context.Context, sizeLimit uint64, wg *sync.WaitGroup) {
+	toOffload, err := lc.registry.EvictLocal(true, sizeLimit)
+	if err != nil {
+		logger.Fatal("error releasing old fractions:", zap.Error(err))
+	}
+	for _, sealed := range toOffload {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			remote, _ := lc.TryOffload(ctx, sealed.instance)
```

should we ignore error here?